cpp_linter/rest_api/github_api.py

```diff
@@ -522,7 +571,178 @@ def _dismiss_stale_reviews(self, url: str):
                         url=f"{url}/{review['id']}/dismissals",
                         method="PUT",
                         data=json.dumps(
-                            {"message": "outdated suggestion", "event": "DISMISS"}
+                            {"message": "Outdated review", "event": "DISMISS"}
                         ),
                         strict=False,
                     )
+
+    def _get_existing_review_comments(self, no_dismissed: bool = True):
+        """Creates the list existing conversation threads to close.
+
+        :param no_dismissed: `True` to ignore any already dismissed comments.
+        """
+        repo_owner, repo_name = self.repo.split("/")
+        query = """
+        query {
+            repository(owner:"%s", name:"%s") {
+                pullRequest(number: %d) {
+                    id
+                    reviewThreads(last: 100) {
+                        nodes {
+                            id
+                            isResolved
+                            isCollapsed
+                            comments(first: 10) {
+                                nodes {
+                                    id
+                                    body
+                                    path
+                                    line
+                                    startLine
+                                    originalLine
+                                    originalStartLine
+                                    author {
+                                        login
+                                    }
+                                    pullRequestReview {
+                                        id
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        """ % (
+            repo_owner,
+            repo_name,
+            self.pull_request,
+        )
+        response = self.api_request(
+            url="https://api.github.com/graphql",
```

Comment:
I believe the URL `https://api.github.com/graphql` could also be abstracted from the REST API URL.
```suggestion
            url=f"{self.api_url}/graphql",
```
This abstraction makes testing with a mock server very easy.